x-pack/plugins/alerting/server/saved_objects/migrations.ts

```diff
@@ -149,21 +160,24 @@ export function getMigrations(
   const migrationRules820 = createEsoMigration(
     encryptedSavedObjects,
     (doc: SavedObjectUnsanitizedDoc<RawRule>): doc is SavedObjectUnsanitizedDoc<RawRule> => true,
-    pipeMigrations(addMappedParams)
+    pipeMigrations(addMappedParams, addSearchType)
   );
 
-  return {
-    '7.10.0': executeMigrationWithErrorHandling(migrationWhenRBACWasIntroduced, '7.10.0'),
-    '7.11.0': executeMigrationWithErrorHandling(migrationAlertUpdatedAtAndNotifyWhen, '7.11.0'),
-    '7.11.2': executeMigrationWithErrorHandling(migrationActions7112, '7.11.2'),
-    '7.13.0': executeMigrationWithErrorHandling(migrationSecurityRules713, '7.13.0'),
-    '7.14.1': executeMigrationWithErrorHandling(migrationSecurityRules714, '7.14.1'),
-    '7.15.0': executeMigrationWithErrorHandling(migrationSecurityRules715, '7.15.0'),
-    '7.16.0': executeMigrationWithErrorHandling(migrateRules716, '7.16.0'),
-    '8.0.0': executeMigrationWithErrorHandling(migrationRules800, '8.0.0'),
-    '8.0.1': executeMigrationWithErrorHandling(migrationRules801, '8.0.1'),
-    '8.2.0': executeMigrationWithErrorHandling(migrationRules820, '8.2.0'),
-  };
+  return mergeSavedObjectMigrationMaps(
+    {
+      '7.10.0': executeMigrationWithErrorHandling(migrationWhenRBACWasIntroduced, '7.10.0'),
+      '7.11.0': executeMigrationWithErrorHandling(migrationAlertUpdatedAtAndNotifyWhen, '7.11.0'),
+      '7.11.2': executeMigrationWithErrorHandling(migrationActions7112, '7.11.2'),
+      '7.13.0': executeMigrationWithErrorHandling(migrationSecurityRules713, '7.13.0'),
+      '7.14.1': executeMigrationWithErrorHandling(migrationSecurityRules714, '7.14.1'),
+      '7.15.0': executeMigrationWithErrorHandling(migrationSecurityRules715, '7.15.0'),
+      '7.16.0': executeMigrationWithErrorHandling(migrateRules716, '7.16.0'),
+      '8.0.0': executeMigrationWithErrorHandling(migrationRules800, '8.0.0'),
+      '8.0.1': executeMigrationWithErrorHandling(migrationRules801, '8.0.1'),
+      '8.2.0': executeMigrationWithErrorHandling(migrationRules820, '8.2.0'),
```

Comment:
The label on the PR says `8.3.0`, so shouldn't this be injected under a `8.3.0` rather than `8.2.0`?

Reply:
I would say, we need migrations for the versions starting from `8.3.0`. Any further `searchSource` modifications should be taken into consideration. That is why `mergeSavedObjectMigrationMaps` is used here.

Reply:
Nice catch, we have moved a new Discover alert to a `8.3.0`. Therefore we need `searchType` param migration of `Es query` alert for `8.3.0`.